components/butterfly/src/server/mod.rs

```diff
@@ -1111,8 +1107,8 @@ mod tests {
             let gossip_port = GOSSIP_PORT.fetch_add(1, Ordering::Relaxed);
             let gossip_listen = format!("127.0.0.1:{}", gossip_port);
             let mut member = Member::default();
-            member.swim_port = (swim_port as i32);
-            member.gossip_port = (gossip_port as i32);
+            member.swim_port = swim_port as i32;
+            member.gossip_port = gossip_port as i32;
```

Comment:
I wonder why these ports are `i32` instead of `u16`

Reply:
That is a fantastic point 🤔 

Our protobuf definitions define this as `int32`. There _is_ a `uint32` [Protobuf scalar type](https://developers.google.com/protocol-buffers/docs/proto#scalar), but no `u16`. We could probably tweak this a bit in our protobuf-to-Rust conversion.

In practice, these types get populated in the first place after being parsed by networking code, so they should be legal port numbers, but we could probably do some more work to ensure that at the type level.